extensions/elasticsearch/elasticsearch-7/src/main/java/com/hazelcast/jet/elastic/ElasticSinks.java

```diff
@@ -0,0 +1,107 @@
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.elastic;
+
+import com.hazelcast.function.FunctionEx;
+import com.hazelcast.function.SupplierEx;
+import com.hazelcast.jet.pipeline.Sink;
+import org.apache.http.HttpHost;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.elasticsearch.action.DocWriteRequest;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.RestHighLevelClient;
+
+import javax.annotation.Nonnull;
+
+import static org.apache.http.auth.AuthScope.ANY;
+
+/**
+ * Provides factory methods for Elasticsearch sinks.
+ * Alternatively you can use {@link ElasticSinkBuilder}
+ *
+ * @since 4.1
+ */
+public final class ElasticSinks {
+
+    private static final int PORT = 9200;
+
+    private ElasticSinks() {
+    }
+
+    /**
+     * Creates an Elasticsearch sink, uses a local instance of Elasticsearch
+     *
+     * @param mapItemFn function that maps items from a stream to an indexing request
+     */
+    public static <T> Sink<T> elastic(@Nonnull FunctionEx<? super T, ? extends DocWriteRequest<?>> mapItemFn) {
+        return elastic(() -> client("localhost", PORT), mapItemFn);
+    }
+
+    /**
+     * Creates an Elasticsearch sink, uses provided clientSupplier and mapItemFn
+     *
+     * @param clientSupplier client supplier
+     * @param mapItemFn      function that maps items from a stream to an indexing request
+     * @param <T>            type of incoming items
+     */
+    public static <T> Sink<T> elastic(
+            @Nonnull SupplierEx<RestHighLevelClient> clientSupplier,
+            @Nonnull FunctionEx<? super T, ? extends DocWriteRequest<?>> mapItemFn
+    ) {
+        return new ElasticSinkBuilder<T>()
+                .clientSupplier(clientSupplier)
+                .mapItemFn(mapItemFn)
+                .build();
+    }
+
+    /**
+     * Returns {@link ElasticSinkBuilder}
+     *
+     * @param <T> type of the items in the pipeline
+     */
+    public static <T> ElasticSinkBuilder<T> builder() {
+        return new ElasticSinkBuilder<T>();
+    }
+
+    static RestHighLevelClient client(String hostname, int port) {
+        return new RestHighLevelClient(
+                RestClient.builder(new HttpHost(hostname, port))
+        );
+    }
+
+    /**
+     * Convenience method to create {@link RestHighLevelClient} with basic authentication and given hostname and port
+     * <p>
+     * Usage:
+     * <pre>
+     *   BatchSource<SearchHit> source = elasticsearch(() -> client("user", "password", "host", 9200));
+     * </pre>
+     */
+    public static RestHighLevelClient client(String username, String password, String hostname, int port) {
```

Comment:
missing annotations. Also this method doesn't provide source/sink, should be moved somewhere else.

Reply:
fixed